src/doc/man/cargo-yank.md

```diff
@@ -15,9 +15,39 @@ The yank command removes a previously published crate's version from the
 server's index. This command does not delete any data, and the crate will
 still be available for download via the registry's download link.
 
-Note that existing crates locked to a yanked version will still be able to
-download the yanked version to use it. Cargo will, however, not allow any new
-crates to be locked to any yanked version.
+Crates should only be yanked in exceptional circumstances, for example, license/copyright issues, accidental
+inclusion of [PII](https://en.wikipedia.org/wiki/Personal_data), credentials, etc... In the case of security
+vulnerabilities, [RustSec](https://rustsec.org/) is typically a less disruptive mechanism to inform users
+and encourage them to upgrade, and avoids the possibility of significant downstream disruption irrespective
+of susceptibility to the vulnerability in question.
+
+Cargo will not use a yanked version for any new project or checkout without a
+pre-existing lockfile, and will generate an error if there are no longer
+any compatible versions for your crate.
+
+For example, the `foo` crate published version `0.22.0` and another crate `bar`
+declared a dependency on version `foo = 0.22`. Now `foo` releases a new, but
+not semver compatible, version `0.23.0`, and finds a critical issue with `0.22.0`.
+If `0.22.0` is yanked, no new project or checkout without an existing lockfile will be
+able to use crate `bar` as it relies on `0.22`.
+
+In this case, the maintainers of `foo` should first publish a semver compatible version
+such as `0.22.1` prior to yanking `0.22.0` so that `bar` and all projects that depend
+on `bar` will continue to work.
```

Comment:
This paragraph concludes the example really well, thought it feels like somewhat a duplicate of line 49-50. A challenge in documentation is putting the right amount. Could we find a way to merge them and make it more concise? It is not a blocker on merging this PR though.